static/js/components/search/NestedAggregatingMenuFilter.js

```diff
@@ -0,0 +1,292 @@
+        TermQuery(filterKey, filterValue)
+      ));
+      return this.fieldContext.wrapFilter(BoolMust(filterTermQueries));
+    } else {
+      return undefined;
+    }
+  }
+
+  /**
+   * Creates the filter for the aggregations of this element. The filter will include (a) any filters that don't apply
+   * to this element's nested path, and (b) filters on this element's nested path minus the filter for this specific
+   * element.
+   */
+  createAggFilter(query) {
+    let filters = [];
+    let nestedPath = this.getNestedPath();
+    let unrelatedFilters = query.getFiltersWithoutKeys(nestedPath);
+    if (unrelatedFilters) {
+      filters.push(unrelatedFilters);
+    }
+    let otherAppliedFiltersOnPath = this.createFilterForOtherElementsOnPath(query);
+    if (otherAppliedFiltersOnPath) {
+      filters.push(NestedQuery(nestedPath, otherAppliedFiltersOnPath));
+    }
+    return filters.length > 0 ? BoolMust(filters) : undefined;
+  }
+
+  /**
+   * Gets an array of all term filters that are applied to this element's same nested path. Returns undefined if no
+   * filters are applied on this element's nested path.
+   *
+   * Example return value: [
+   *   {'term': {'program.enrollments.course_title': 'Some Course Title'},
+   *   {'term': {'program.enrollments.payment_status': 'Paid'}
+   * ]
+   */
+  getAllTermFiltersOnPath(query) {
+    let nestedPath = this.getNestedPath();
+    let appliedNestedFilters = query.getFiltersWithKeys(nestedPath);
+    let filterElement = R.pathOr({}, ['nested', 'filter'], appliedNestedFilters);
+    if (R.path(['bool', 'must'], filterElement)) {
+      return R.path(['bool', 'must'], filterElement);
+    } else if (filterElement.term) {
+      return [filterElement];
+    } else {
+      return [];
+    }
+  }
+
+  /**
+   * Creates an 'AND' filter for all filters that are applied on this element's nested path, minus the filter for
+   * this specific element (if it exists). If no other filters are applied on this element's nested path, undefined is
+   * returned.
+   *
+   * Example return value: {
+   *   'bool': {
+   *     'must': [
+   *       {'term': {'program.enrollments.course_title': 'Some Course Title'}
+   *     ]
+   *   }
+   * }
+   */
+  createFilterForOtherElementsOnPath(query) {
+    let allTermFilters = this.getAllTermFiltersOnPath(query);
+    let otherTermFilters = [];
+    if (allTermFilters.length > 0) {
+      // Filter out term filters for this element
+      otherTermFilters = R.filter(
+        R.compose(
+          R.not,
+          R.equals(this.key),
+          R.head,
+          R.keys,
+          R.prop('term')
+        )
+      )(allTermFilters);
+    }
+    return otherTermFilters.length > 0 ? BoolMust(otherTermFilters) : undefined;
+  }
+
+  /**
+   * Gets the appropriate terms bucket for this element's agg query.
+   */
+  getTermsBucket(query) {
+    let otherAppliedFiltersOnPath = this.createFilterForOtherElementsOnPath(query);
```

Comment:
Why are we adding the filter here and in `createAggFilter`? Wouldn't the filter be added to the aggregations if it exists in `filters` and `filtersMap`?